src/symbanded/symbanded.jl

```diff
@@ -164,11 +154,33 @@ function eigvals!(A::Symmetric{T,<:BandedMatrix{T}}, B::Symmetric{T,<:BandedMatr
     eigvals!(A)
 end
 
-function eigvals(A::Symmetric{<:Any,<:BandedMatrix}, B::Symmetric{<:Any,<:BandedMatrix})
-    AA = if bandwidth(A) >= bandwidth(B)
+# Internal function similar to copyto! to overwrite the bands of A with that of B
+function copyto_bandedsym!(A::Symmetric{<:Number,<:BandedMatrix}, B::Symmetric{<:Number,<:BandedMatrix})
```

Comment:
I don't understand how this is different than `copyto!`.

Reply:
This is equivalent to `copyto!` for matrices having the same size and the same `uplo`. This may be seen as a restricted version of `copyto!`, which is all we need for `eigen` and `eigvals`. A `copyto!` implementation should be more general, but this is broken currently (https://github.com/JuliaLang/julia/issues/45958). `copyto!` also performs a linear copy, so while the results might be the same, the idea of copying bands is different from what `copyto!` does.

Reply:
Let's also see what's the decision on https://github.com/JuliaLang/julia/issues/46005, since `copyto!` isn't consistently implemented in `LinearAlgebra` either

Reply:
I would be in favour of not supporting `copyto!` when the axes don't match....

Reply:
I have updated this to `copyto!` now, with the restriction that sizes must match. I have added a path for the case where the `uplo` doesn't match, but this isn't optimized yet. However, this may be optimized in a separate PR.